src/quo2/components/drawers/action_drawers.cljs

```diff
@@ -0,0 +1,78 @@
+(ns quo2.components.drawers.action-drawers
+  (:require  [status-im.ui.components.react :as react]
+             [quo.react-native :as rn]
+             [quo2.components.markdown.text :as text]
+             [quo2.components.icon :as icon]
+             [status-im.ui.components.list.views :as list]
+             [quo2.foundations.colors :as colors]))
+
+(defn- get-icon-color [color]
+  (if (= color :red)
+    (colors/theme-colors colors/danger-50 colors/danger-60)
+    (colors/theme-colors colors/neutral-50 colors/neutral-40)))
+
+(defn action [color]
+  (fn [{:keys [icon label sub-label right-icon on-press]}]
+    [rn/touchable-opacity {:on-press on-press}
+     [react/view {:style
+                  {:flex 1
+                   :height (if sub-label 56 47)
+                   :margin-horizontal 20
+                   :flex-direction :row}}
+      [react/view {:style
+                   {:height 20
+                    :margin-top :auto
+                    :margin-bottom :auto
+                    :margin-right 12
+                    :width 20}}
+       [icon/icon icon
+        {:color (get-icon-color color)
+         :size 20}]]
+      [react/view
+       {:style
+        {:flex 1
+         :justify-content :center}}
+       [text/text
+        {:size :paragraph-1
+         :weight :medium
+         :style
+         {:color (when (= color :red)
+                   (colors/theme-colors colors/danger-50 colors/danger-60))}}
+        label]
+       (when sub-label [text/text
+                        {:size :paragraph-2
+                         :style
+                         {:color (colors/theme-colors colors/neutral-50 colors/neutral-40)}}
+                        sub-label])]
+      (when right-icon
+        [react/view {:style
+                     {:height 20
+                      :margin-top :auto
+                      :margin-bottom :auto
+                      :width 20}}
+         [icon/icon right-icon
+          {:color (get-icon-color color)
+           :size 20}]])]]))
+
+(defn action-drawer [{:keys [actions actions-with-consequence]}]
+  [react/view {:style
+               {:flex 1}}
+   [list/flat-list
+    {:key-fn (fn [_ index] index)
+     :list-key "actions-"
+     :data                              actions
+     :render-fn                         (action :black)}]
```

Comment:
It'd be even better if we could use semantic colors (primary, secondary, etc) instead of actual color names.

Reply:
I don't really know what those semantic colors are. It's just about whether it's top section or bottom. I guess I can change to :top and :bottom 🤔

Reply:
Sounds good to me @J-Son89. Imagine when (not if) the design team changes from red to orange, then the code won't make much sense.

Reply:
yep, I completely agree :)